app/controllers/users.js

```diff
@@ -107,54 +108,96 @@ exports.create = function(req, res) {
   }
 };
 
+exports.createJwt = function (req, res) {
+  if (req.body.name && req.body.password && req.body.email && req.body.username) {
+    User.findOne({
+      email: req.body.email
+    })
+      .exec()
+      .then(function (err, existingUser) {
+        if (err) {
+          return res.status(409).json({
+            message: "Email already exists"
+          });
+        }
+        if (!existingUser) {
+          var user = new User(req.body);
+          user.avatar = avatars[user.avatar];
+          user.save(function (error, newUser) {
+            if (error) {
+              return res.status(400).json({
+                message: "Signup not successful"
```

Comment:
This should be a server error not a 400 error.

Reply:
Noted!